src/libs/antares/benchmarking/info_collectors.cpp

```diff
@@ -0,0 +1,152 @@
+			{
+				auto& cluster = i->second;
+				if (cluster->enabled)
+					nbEnabledThermalClusters++;
+			}
+		}
+
+		// Adding an item related to number of enabled thermal clusters to the file content
+		file_content.addItemToSection("from study", "enabled thermal clusters", nbEnabledThermalClusters);
+	}
+
+	void StudyInfoCollector::enabledBindingConstraintsCountToFileContent(FileContent& file_content)
+	{
+		unsigned int nbEnabledBC = study_.runtime->bindingConstraintCount;
+		unsigned int nbEnabledHourlyBC(0), nbEnabledDailyBC(0), nbEnabledWeeklyBC(0);
+
+		for (uint i = 0; i < nbEnabledBC; i++)
+		{
+			switch (study_.runtime->bindingConstraint[i].type)
+			{
+			case BindingConstraint::Type::typeHourly:
+				nbEnabledHourlyBC++;
+				break;
+			case BindingConstraint::Type::typeDaily:
+				nbEnabledDailyBC++;
+				break;
+			case BindingConstraint::Type::typeWeekly:
+				nbEnabledWeeklyBC++;
+				break;
+			default:
+				break;
+			}
+		}
+
+		file_content.addItemToSection("from study", "enabled BC", nbEnabledBC);
+		file_content.addItemToSection("from study", "enabled hourly BC", nbEnabledHourlyBC);
+		file_content.addItemToSection("from study", "enabled daily BC", nbEnabledDailyBC);
+		file_content.addItemToSection("from study", "enabled weekly BC", nbEnabledWeeklyBC);
+	}
+
+	void StudyInfoCollector::unitCommitmentModeToFileContent(FileContent& file_content)
+	{
+		const char* unitCommitment = UnitCommitmentModeToCString(study_.parameters.unitCommitment.ucMode);
+		file_content.addItemToSection("from study", "unit commitment", unitCommitment);
+	}
+
+	void StudyInfoCollector::maxNbYearsInParallelToFileContent(FileContent& file_content)
+	{
+		file_content.addItemToSection("from study", "max parallel years", study_.maxNbYearsInParallel);
+	}
+
+	void StudyInfoCollector::solverVersionToFileContent(FileContent& file_content)
+	{
+		// Example : 8.3.0 -> 830
+		const unsigned int version
+			= 100 * ANTARES_VERSION_HI + 10 * ANTARES_VERSION_LO + ANTARES_VERSION_BUILD;
+
+		file_content.addItemToSection("from study", "antares version", version);
+	}
+
+	// Collecting data from simulation
+	// ---------------------------------
+	void SimulationInfoCollector::toFileContent(FileContent& file_content)
+	{
+		file_content.addItemToSection("from simulation", "variables", opt_info_.nbVariables);
+		file_content.addItemToSection("from simulation", "constraints", opt_info_.nbConstraints);
+		file_content.addItemToSection("from simulation", "non-zero coefficients", opt_info_.nbNonZeroCoeffs);
+	}
+
+	// Collecting durations from simulation
+	// -------------------------------------
+	void DurationCollector::toFileContent(FileContent& file_content)
+	{		
+		for (pair<string, vector<int64_t>> element : duration_items_)
+		{
+			std::string name = element.first;
+			vector<int64_t> durations = element.second;
+			int64_t duration_sum = 0;
+			for (auto& duration : durations)
+			{
+				duration_sum += duration;
+			}
```

Comment:
Use `auto&` to avoid unnecessary copies. Also, you could keep only the sum of durations, removing the need for loop at lines 137-140.

Reply:
I understand. But if we wanted to know the number of calls, we could just increment a counter. On the other hand, it doesn't really matter since we don't expect large vectors, so memory usage won't be an issue.